src/probability/kernel/disintegration.lean

```diff
@@ -0,0 +1,407 @@
+    change (ρ.fst) {a : α | a ∉ {a' : α | cond_kernel_real ρ' a' (range f) = 1}} = 0,
+    rw [← ae_iff, ← h_fst],
+    refine ae_cond_kernel_real_eq_one ρ' hf.measurable_set_range _,
+    rw measure.map_apply h_prod_embed.measurable,
+    swap, { exact measurable_snd hf.measurable_set_range.compl, },
+    convert measure_empty,
+    ext1 x,
+    simp only [mem_compl_iff, mem_range, preimage_set_of_eq, prod_map, mem_set_of_eq,
+      mem_empty_iff_false, iff_false, not_not, exists_apply_eq_apply], },
+  classical,
+  obtain ⟨x₀, hx₀⟩ : ∃ x, x ∈ range f := range_nonempty _,
+  let η' := kernel.piecewise hm (cond_kernel_real ρ')
+    (kernel.deterministic (λ _, x₀) measurable_const),
+  -- We show that `kernel.comap_right η' hf` is a suitable Markov kernel.
+  refine ⟨kernel.comap_right η' hf, _, _⟩,
+  { refine kernel.is_markov_kernel.comap_right _ _ (λ a, _),
+    rw kernel.piecewise_apply',
+    split_ifs with h_mem h_not_mem,
+    { exact h_eq_one_of_mem _ h_mem, },
+    { rw [kernel.deterministic_apply' _ _ hf.measurable_set_range, set.indicator_apply,
+        if_pos hx₀], }, },
+  have : kernel.const γ ρ = kernel.comap_right (kernel.const γ ρ') h_prod_embed,
+  { ext c t ht : 2,
+    rw [kernel.const_apply, kernel.comap_right_apply' _ _ _ ht, kernel.const_apply,
+      measure.map_apply h_prod_embed.measurable (h_prod_embed.measurable_set_image.mpr ht)],
+    congr' with x : 1,
+    rw ← @prod.mk.eta _ _ x,
+    simp only [id.def, mem_preimage, prod.map_mk, mem_image, prod.mk.inj_iff, prod.exists],
+    refine ⟨λ h, ⟨x.1, x.2, h, rfl, rfl⟩, _⟩,
+    rintros ⟨a, b, h_mem, rfl, hf_eq⟩,
+    rwa hf.injective hf_eq at h_mem, },
+  rw [this, kernel.const_eq_comp_prod_real ρ'],
+  ext c t ht : 2,
+  rw [kernel.comap_right_apply' _ _ _ ht,
+    kernel.comp_prod_apply _ _ _ (h_prod_embed.measurable_set_image.mpr ht), kernel.const_apply,
+    h_fst, kernel.comp_prod_apply _ _ _ ht, kernel.const_apply],
+  refine lintegral_congr_ae _,
+  filter_upwards [h_ae] with a ha,
+  rw [kernel.prod_mk_left_apply', kernel.prod_mk_left_apply', kernel.comap_right_apply'],
+  swap, { exact measurable_prod_mk_left ht, },
+  have h1 : {c : ℝ | (a, c) ∈ prod.map id f '' t} = f '' {c : Ω | (a, c) ∈ t},
+  { ext1 x,
+    simp only [prod_map, id.def, mem_image, prod.mk.inj_iff, prod.exists, mem_set_of_eq],
+    split,
+    { rintros ⟨a', b, h_mem, rfl, hf_eq⟩,
+      exact ⟨b, h_mem, hf_eq⟩, },
+    { rintros ⟨b, h_mem, hf_eq⟩,
+      exact ⟨a, b, h_mem, rfl, hf_eq⟩, }, },
+  have h2 : cond_kernel_real ρ' (c, a).snd = η' (c, a).snd,
+  { rw [kernel.piecewise_apply, if_pos ha], },
+  rw [h1, h2],
+end
+
+variables [nonempty Ω]
+
+/-- Conditional kernel of a measure on a product space: a Markov kernel such that
+`ρ : measure (α × Ω) = (ρ.fst : measure α) ⊗ₖ (cond_kernel ρ : kernel α Ω)` (up to augmentations
+of the spaces to make that expression valid: see `probability_theory.measure_eq_comp_prod`). -/
```

Comment:
I'm not sure I understand the comment between parenthesis here (you are not augmenting the space in this definition). Can you remove it or expand it?

Reply:
I wanted to explain that the equality I wrote in that docstring was simplified and did not quite make sense (the comp-prod of kernels does not take a measure as first argument, for example). I replaced it with the exact result.